rebase: update kubernetes to v1.23.0

Madhu-1 opened the pull request:
updating go dependency to latest kubernetes released version i.e v1.23.0

closes https://github.com/ceph/ceph-csi/issues/2698
closes #2590 
depends-on: #2703

note:- other dependencies are auto-updated when I ran `go mod` commands

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
rebase: update kubernetes to v1.23.0

updating go dependency to latest kubernetes
released version i.e v1.23.0

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

nixpanic commented on go.mod:
```diff
@@ -22,20 +22,21 @@ require (
-	k8s.io/cloud-provider v0.22.4
-	k8s.io/klog/v2 v2.10.0
+	k8s.io/cloud-provider v0.23.0
+	k8s.io/klog/v2 v2.30.0
```
I do not think you can update `klog` yet, there is an issue with `controller-runtime`?

Madhu-1 replied:
Yes, correct. reverted this one but facing other issues. reworking on this one will update this once everything works locally.

Madhu-1 replied:
controller runtime is updated to kube 1.23.0 in master. updated controller runtime to fix the klog issue.

Madhu-1 committed:
ci: ignore go.sum from codespell check

as go-sum is auto generated, ignore checking
codespell on this file.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: update golang dependencies to 1.17.5 release

updating the depencendices to golang 1.17.5
release

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: update action go.mod to 1.17

As we are updating the cephcsi main repo
to golang 1.17, updating the action package
to golang 1.17

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: install go version specified in build.env

installed the go version specified in build.env
in the test docker image.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: install findutils in dockerfile.test

currently getting find command not found
and xargs command not found when we run
the dockerfile.test. installing findutils
to fix it.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/retest all

nixpanic reviewed (changes requested):
Looks good to me, but something causes building `e2e.test` to fail now:
```
go test -tags=pacific -mod=vendor -c ./e2e
# sigs.k8s.io/json/internal/golang/encoding/json
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
```

Madhu-1 commented:
>+ ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@n61.gusty.ci.centos.org 'cd /opt/build/go/src/github.com/ceph/ceph-csi && make containerized-build CONTAINER_CMD=podman TARGET=e2e.test ENV_CSI_IMAGE_NAME=registry-ceph-csi.apps.ocp.ci.centos.org/ceph-csi USE_PULLED_IMAGE=yes'
Warning: Permanently added 'n61.gusty.ci.centos.org,172.19.2.189' (ECDSA) to the list of known hosts.
cephcsi image settings: registry-ceph-csi.apps.ocp.ci.centos.org/ceph-csi version canary
podman inspect -f '{{.Id}}' registry-ceph-csi.apps.ocp.ci.centos.org/ceph-csi:devel > .devel-container-id

@nixpanic CI is failing because it's using the older devel image which is built for golang 1.16 we need a new image to be pushed to the CI registry for CI to pass. can you please help with that

Rakshith-R commented on go.mod:
```diff
@@ -22,21 +21,133 @@ require (
+	k8s.io/kubernetes v1.23.0
+	k8s.io/mount-utils v0.23.0
+	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
+	sigs.k8s.io/controller-runtime v0.11.0-beta.0.0.20211208212546-f236f0345ad2
```
```suggestion
	sigs.k8s.io/controller-runtime v0.11.0
```
let's wait till we have a controller-runtime stable release ?

Madhu-1 replied:
we don't know when we will have the next stable release. and more over as we are sticking to a commit i think we are good we are already sticking to a commit in multiple places.

nixpanic replied:
I think this is fine. When controller-runtime releases a new version, dependabot will send a PR to update it.

nixpanic commented:
> 

I think images are rebuilt after a PR has been merged. See https://jenkins-ceph-csi.apps.ocp.ci.centos.org/job/build-images/

However, if a PR changes the build.env file, it should not download the pre-built container image, but build a new one from scratch... If that is not working for some reason, we need to fix that.

nixpanic commented:
> > 
> 
> I think images are rebuilt after a PR has been merged. See https://jenkins-ceph-csi.apps.ocp.ci.centos.org/job/build-images/
> 
> However, if a PR changes the build.env file, it should not download the pre-built container image, but build a new one from scratch... If that is not working for some reason, we need to fix that.

Ah, it seems this is not done for all CI jobs...

https://github.com/ceph/ceph-csi/blob/d351e009928e3fd5b924b81a9328e19167c1624f/containerized-tests.groovy#L96-L102

It might be easier to send a PR updating Golang, get that merged so the container-images are rebuilt. And then update the Kubernetes dependency.

Madhu-1 commented:
https://github.com/ceph/ceph-csi/pull/2703 is to update the golang version to 1.17 once it's merged will rebase this one.

nixpanic commented:
When this PR is merged, #2590 can be closed as well.

nixpanic commented on build.env:
```diff
@@ -16,7 +16,7 @@ BASE_IMAGE=docker.io/ceph/ceph:v16
 
 # standard Golang options
-GOLANG_VERSION=1.16.4
+GOLANG_VERSION=1.17.4
```
This obviously is part of #2703 already, so a rebase will be needed. Setting `depends-on: #2703` in the description so that Mergify should pick it up automatically.

nixpanic commented on scripts/Dockerfile.test:
```diff
@@ -38,6 +39,9 @@ RUN source /build.env \
     && dnf -y clean all \
     && gem install mdl \
+    && mkdir -p ${GOROOT} \
+    && curl https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-amd64.tar.gz  \
```
This makes me sad, it breaks on non-amd64 architectures... But I also do not know of a better way to do this, current Fedora 35 (`:latest`) has golang-1.16.x :disappointed:

Madhu-1 replied:
yes, I borrowed it from https://github.com/ceph/ceph-csi/blob/devel/scripts/Dockerfile.devel#L15. https://github.com/ceph/ceph-csi/issues/2707 is for tracking

Madhu-1 commented:
@Rakshith-R you have requested changes. PTAL

Madhu-1 committed:
cleanup: remove k8s.io/utils from vendor

as part of running `go mod tidy` the unused
k8s.io/utils is removed from the vendor
directory.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
>X Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute

CI failure

Madhu-1 commented:
/retest ci/centos/upgrade-tests-rbd

Madhu-1 commented:
/retest ci/centos/upgrade-tests-cephfs

Madhu-1 commented:
>X Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute

failed due to above error

Madhu-1 commented:
@Rakshith-R can you please bring your approval back on this one?
